src/components/submit-game/submit-game.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, ViewChild, Pipe, PipeTransform } from '@angular/core';
+import { MatSnackBar } from '@angular/material';
+import { Router } from '@angular/router';
+
+import { User, GameType, Game } from '../../datatypes/datatypes';
+import { BackendService } from '../../services/backend.service';
+import { SessionService } from '../../services/session.service';
+
+@Component({
+  selector: 'submit-game',
+  templateUrl: './submit-game.component.html',
+  styleUrls: ['./submit-game.component.less']
+})
+export class SubmitGameComponent {
+
+  players: User[];
+  player: number;
+  playmate: number;
+  buttonsDisabled: boolean[] = new Array(4);
+
+  constructor(private snackBar: MatSnackBar, private backend: BackendService, private router: Router, private session: SessionService) {
+    this.players = this.session.getPlayers();
+  }
+
+  /** Checks if provided information is correct and formats it correct */
+  submitGame(event: Event, gameType, schwarz, points, runners, virgins): void {
+    if (event instanceof KeyboardEvent && event.keyCode !== 13) return; // keyCode 13: Enter key
+    if (gameType && schwarz && points && runners && virgins && gameType.value) {
+
+      if (gameType.value == "Ramsch") {
+        if (virgins.value) {
+          if (virgins.value < 0 || virgins.value > 3) {
+            this.snackBar.open("Bitte überprüfen Sie Ihre Eingaben auf Richtigkeit.");
+            return;
+          }
+          const game: Game = {
+            participants: this.players.map(p => p.id),
+            player: this.player,
+            playmate: this.playmate,
+            gameType: (<any>GameType)[gameType.value],
+            schwarz: false,
+            points: 0,
```

Comment:
I think points should be counted in Ramsch for stats (average points when lost)
Maybe in the future of all participants

Otherwise I would set it to -1

Reply:
Yeah they should, but the submission logic is not in the scope of this PR

Reply:
Then I would change it to -1 temporary

See #12 for the future